File: server/internal/domain/subscription.go
package domain

import "time"

type SubscriptionPlanStatus int

const (
	SubscriptionPlanDisabled SubscriptionPlanStatus = 0
	SubscriptionPlanEnabled  SubscriptionPlanStatus = 1
)

type SubscriptionPlan struct {
	ID             uint64                 `gorm:"primaryKey" json:"id"`
	Name           string                 `gorm:"size:128;not null" json:"name"`
	QuotaValue     int64                  `gorm:"not null" json:"quota_value"`
	PriceCents     int64                  `gorm:"not null" json:"price_cents"`
	Currency       string                 `gorm:"size:8;not null;default:'CNY'" json:"currency"`
	IntervalMonths int                    `gorm:"not null;default:1" json:"interval_months"`
	Status         SubscriptionPlanStatus `gorm:"not null;default:1" json:"status"`
	SortOrder      int                    `gorm:"not null;default:0" json:"sort_order"`
	CreatedAt      time.Time              `json:"created_at"`
	UpdatedAt      time.Time              `json:"updated_at"`
}

func (SubscriptionPlan) TableName() string { return "subscription_plans" }

type UserSubscriptionStatus int

const (
	UserSubscriptionInactive  UserSubscriptionStatus = 0
	UserSubscriptionActive    UserSubscriptionStatus = 1
	UserSubscriptionCancelled UserSubscriptionStatus = 2
	UserSubscriptionExpired   UserSubscriptionStatus = 3
)

type UserSubscription struct {
	ID           uint64                 `gorm:"primaryKey" json:"id"`
	UserID       uint64                 `gorm:"not null;index" json:"user_id"`
	PlanID       uint64                 `gorm:"not null;index" json:"plan_id"`
	Status       UserSubscriptionStatus `gorm:"not null;default:1" json:"status"`
	StartedAt    time.Time              `gorm:"not null" json:"started_at"`
	ExpiresAt    time.Time              `gorm:"not null;index" json:"expires_at"`
	LastBilledAt *time.Time             `json:"last_billed_at,omitempty"`
	CreatedAt    time.Time              `json:"created_at"`
	UpdatedAt    time.Time              `json:"updated_at"`
}

func (UserSubscription) TableName() string { return "user_subscriptions" }

File: admin/src/api/subscription.ts
import { client } from './client'
import type { ApiResponse, PagedResponse } from '../types/api'

export interface SubscriptionPlan {
  id: number
  name: string
  quota_value: number
  price_cents: number
  currency: string
  interval_months: number
  status: number
  sort_order: number
  created_at: string
  updated_at: string
}

export interface SubscriptionPlanForm {
  name: string
  quota_value: number
  price_cents: number
  currency: string
  interval_months: number
  sort_order: number
}

export interface UserSubscription {
  id: number
  user_id: number
  plan_id: number
  status: number
  started_at: string
  expires_at: string
  last_billed_at?: string
  created_at: string
  updated_at: string
}

export function listSubscriptionPlans(page = 1, pageSize = 20) {
  return client.get<ApiResponse<PagedResponse<SubscriptionPlan>>, ApiResponse<PagedResponse<SubscriptionPlan>>>(
    `/subscriptions/plans?page=${page}&page_size=${pageSize}`,
  )
}

export function createSubscriptionPlan(data: SubscriptionPlanForm) {
  return client.post<ApiResponse<SubscriptionPlan>, ApiResponse<SubscriptionPlan>>('/subscriptions/plans', data)
}

export function listUserSubscriptions(page = 1, pageSize = 20, userId?: number) {
  let url = `/subscriptions?page=${page}&page_size=${pageSize}`
  if (userId) {
    url += `&user_id=${userId}`
  }
  return client.get<ApiResponse<PagedResponse<UserSubscription>>, ApiResponse<PagedResponse<UserSubscription>>>(url)
}

export function updateSubscriptionPlan(id: number, data: Partial<SubscriptionPlanForm> & { status?: number }) {
  return client.put<ApiResponse<SubscriptionPlan>, ApiResponse<SubscriptionPlan>>(`/subscriptions/plans/${id}`, data)
}

export function deleteSubscriptionPlan(id: number) {
  return client.delete<ApiResponse<unknown>, ApiResponse<unknown>>(`/subscriptions/plans/${id}`)
}

/** 为用户开通订阅 */
export function subscribeUser(userId: number, planId: number) {
  return client.post<ApiResponse<UserSubscription>, ApiResponse<UserSubscription>>('/subscriptions', {
    user_id: userId,
    plan_id: planId,
  })
}
